Game4.py

```diff
@@ -0,0 +1,165 @@
+                    if event.key == pygame.K_c:
+                        new_obstical = True
+                        obstical_height = obs_y-50
+
+            # check keyboard for main player
+            pressed = pygame.key.get_pressed()
+            # up to jump
+            if pressed[pygame.K_UP] and P1_stamina_bar.bars >= player.jumpcost:
+                if  player.play_y == (360 - player.play_len):
+                    P1_stamina_bar.decreaseBar(player.jumpcost)
+                    player.jump()
+                    P1_stamina_bar.draw()
+            # left to move left
+            if pressed[pygame.K_LEFT]:
+                player.moveLeft()
+            # right to move right
+            if pressed[pygame.K_RIGHT]:
+                player.moveRight()
+
+            # refresh screen
+            self.screen.fill(colors['BLACK'])
+
+            # update and draw player
+            player.update()
+            player.draw()
+
+            # update current time
+            current_time = pygame.time.get_ticks()
+
+            # generate obstacle at random time
+            if (current_time - prev_time > obs_dt):
+                # uncomment below to generate random obstacles:
+                # obstacles.append(Obstacle(obs_x, obs_y, obs_len, self.screen,'BLUE'))
+                new_obstical = False
+                prev_time = current_time
+                obs_dt = randint(1000, 3000)
+            # generate obstacle player's obstacle at appropriate height
+            if (new_obstical == True and P2_stamina_bar.bars >= 33):
+                new_obstical = False
+                P2_stamina_bar.decreaseBar(33)
+                obstacles.append(Obstacle(obs_x, obstical_height, obs_len, self.screen,'RED'))
+
+            # move each obstacle forward
+            for obstacle in obstacles:
+                obstacle.moveForward()
+                # check for collision between player and obstacles
+                if player.isCollide(obstacle.obs_x, obstacle.obs_y, obstacle.obs_len):
+                    self.gameOver(str(count))
+            # remove obstacle from list if off screen
+            obstacles = [obstacle for obstacle in obstacles if not obstacle.isGone()]
+
+            # update stamina bars
+            P1_stamina_bar.increaseBar()
+            P2_stamina_bar.increaseBar(1.5)
+            P1_stamina_bar.draw()
+            P2_stamina_bar.draw()
+
+            # display score
+            font = pygame.font.SysFont("comicsansms", 72)
+            text = font.render(str(count), True, (255,255,255))
+            self.screen.blit(text,[200,10])
+
+            # update screen
+            pygame.display.flip()
+            self.clock.tick(30)
+
+
+    def gameOver(self,score):
+        """Game over screen."""
+        # main event loop
+        while 1:
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    sys.exit()
```

Comment:
Going beyond: you can also use either of the following. Experiment with these styles, and see if they're to your liking.

```python
if any(event.type == pygame.QUIT for event in pygame.event.get()):
    sys.exit()
```

```python
if pygame.QUIT  in {event.type event in pygame.event.get()}:
    sys.exit()
```

(`{…}` makes a set; same as `set(…)`.)